e2e/cypress/integration/enterprise/system_console/team_scheme_spec.js

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+import {channelUrl, setUserTeamAndChannelMemberships, enablePermission, removePermission, channelMentionsPermissionCheck, resetPermissionsToDefault} from './system_scheme_spec.js';
+
+export const deleteExistingTeamOverrideSchemes = () => {
+    cy.apiLogin('sysadmin');
+    cy.apiGetSchemes('team').then((res) => {
+        res.body.forEach((scheme) => {
+            cy.apiDeleteScheme(scheme.id);
+        });
+    });
+};
+
+describe('Team Scheme Permissions Test', () => {
+    before(() => {
+        // * Check if server has license
+        cy.requireLicense();
+
+        // Reset permissions in system scheme to defaults.
+        resetPermissionsToDefault();
+
+        // Delete any existing team override schemes
+        deleteExistingTeamOverrideSchemes();
+
+        // # Visit the permissions page
+        cy.visit('/admin_console/user_management/permissions');
+    });
+
+    it('MM-23018 - Create a team override scheme', () => {
+        // # Visit the permissions page
+        cy.visit('/admin_console/user_management/permissions/team_override_scheme');
+
+        // # Give the new team scheme a name
+        cy.get('#scheme-name').type('Test Team Scheme');
+
+        // # Assign the new team scheme to the eligendi team using the add teams modal
+        cy.findByTestId('add-teams').click();
+
+        cy.get('#selectItems').type('eligendi');
+
+        cy.get('.team-info-block').then((el) => {
+            el.click();
+        });
+
+        cy.get('#saveItems').click();
+
+        // # Save config
+        cy.get('#saveSetting').click();
+
+        // * Ensure that the team scheme was created and assigned to the team
+        cy.findByTestId('permissions-scheme-summary').within(() => {
+            cy.get('.permissions-scheme-summary--header').should('include.text', 'Test Team Scheme');
+            cy.get('.permissions-scheme-summary--teams').should('include.text', 'eligendi');
+        });
+    });
```

Comment:
Since this is a generic testcase, you can leave this test here in this file `team_scheme_spec.js` as it makes more sense and can move the other test to a different file `channel_mentions_team_scheme_spec.js`

Reply:
my assumption was that eventually we would add more permission tests to this same file - splitting out half the tests into a different file would result in a bunch of duplication - thoughts @srkgupta?

Reply:
I decided to keep this test case in `use_channel_mentions_scheme_spec.js` since it is required as setup for the `use_channel_mentions` part of the test - maybe later it can be extracted into a separate file